### ISSUE TITLE
Image layer rotate initialization with a rotation matrix

### ISSUE THREAD

**Issue by thewtex** [napari/napari#2628]
## 🚀 Feature

When creating an `Image` layer with its constructor or the `add_image` kwargs, enable passing a rotation matrix to initialize the `rotate` attribute. Currently, only rotation angles can be passed. 

## Motivation

This will avoid the need to convert a rotation matrix into angles.

Use case: https://github.com/InsightSoftwareConsortium/itk-napari-conversion/issues/7
Use case: https://github.com/InsightSoftwareConsortium/napari-itk-io/blob/83076c14a276d450cf4fdde87bc1b2d56eaaa4ec/napari_itk_io/_reader.py#L98-L104

## Pitch

It looks like the angles are converted into a rotation matrix, checking the value of `image_layer.rotate` after construction.

## Additional context

We could switch on `rotate.ndim` -- 1 means angles, 2 means rotation matrix.

**Comment by jni** [comment]
@thewtex As an immediate workaround, could you pass the rotation matrix as an affine matrix?

But, since the switch can be perfectly unambiguous here, I'm +1 on this proposal.

**Comment by thewtex** [comment]
Hey @jni ! Thanks for the note and support! Currently, we are providing `scale` and `translate`, which would be awesome to continue to use because we can pull them back out in the same form, easily.

The rotation as a matrix would also help for the `add_image` API per:

 https://github.com/InsightSoftwareConsortium/napari-itk-io/blob/83076c14a276d450cf4fdde87bc1b2d56eaaa4ec/napari_itk_io/_reader.py#L98-L104

**Comment by sofroniewn** [comment]
Thanks for the links here @thewtex, the concrete use cases will help. What you propose makes sense too. We've actually got a couple other issues like #2163 with our transforms that we need to address too (right now having a scale/ translate will get clobbered by having an affine, curious what your thoughts on that one are, and your expectations for rotation!). Maybe we can then take another look at our transformation API more generally and fix this rotation issue along with those other ones that have come up. I think @tlambert03 and I have discussed some of this before too

**Comment by thewtex** [comment]
@sofroniewn thank you -- these improvements would be very helpful!

I followed up in #2163, but for elastix, it would be helpful to avoid having scale / translate clobbered by an affine result. This is because elastix assumes that the input images have a scale / translate associated with them always (this is pixel *spacing* and image *origin* in itk-terminology). If we keep these separate and do not compose them, they can be read in with the [napari-itk-io](https://github.com/InsightSoftwareConsortium/napari-itk-io) plugin, then passed to elastix in the napari-elastix plugin. If napari supports affine transforms in addition to the others, @NHPatterson will be able visualize an elastix-transformed result in napari without resampling. In sum,

```
affine * (rotate * scale + translate)
```

would serve most use cases.

**Comment by sofroniewn** [comment]
Nice, I responded over in #2163 too, composing sounds like it will be a big improvement for everyone. We can work on this soon. Will keep you posted when we have a PR in progress.

**Comment by andy-sweet** [comment]
@thewtex : do you have an example that reproduces this issue?

I used the issue described in #2163 as a base and modified it to create the following example, which works at head.

```python

import numpy as np
import napari
from napari.utils import transforms
from skimage.data import astronaut
from skimage.transform import rescale, rotate
from skimage.color import rgb2gray

# load in RGB image -> gray
# physical size is arbitrarily 1 units/px
full_size_im = rgb2gray(astronaut())

# resize image by half
# physical size is 2 units/px
half_size_im = rescale(full_size_im, (0.5, 0.5), preserve_range=True)

# rotate full size 45 degrees
full_size_im_rot45 = rotate(full_size_im, 45, True, preserve_range=True)

# rigid transformation found through elastix, in physical coords
# converted, inverted and reordered
# using code & ideas from https://github.com/image-transform-converters/image-transform-converters
rotate_mat = np.array(
    [
        [0.70711116, -0.7071024],
        [0.7071024, 0.70711116],
    ]
)
translate_vec = np.array([361.34715411, 0.70937201])

viewer = napari.Viewer()
viewer.add_image(
        full_size_im_rot45,
        name="full rot 45 resize",
        scale=[1, 1],
        colormap="viridis",
        opacity=0.5,
        )
viewer.add_image(
        half_size_im,
        name="half registered",
        translate=translate_vec,
        scale=[2, 2],
        rotate=rotate_mat,
        colormap="red",
        )
print(viewer.layers["half registered"].affine.affine_matrix)
```

The `rotate` matrix value should eventually be handled in [`compose_linear_matrix`](https://github.com/napari/napari/blob/46f295b599269bc79485d6d7febf1265a4fc7c65/napari/utils/transforms/transform_utils.py#L70), if it's not a scalar or 1D.

To make this example, I split the 3x3 2D rigid transform matrix into a 2x2 rotation matrix and 2x1 translation vector and passed these through to `add_image`'s `translate` and `rotate` arguments.

In general, you should be able to pass through an n-dimensional rotation matrix (assuming the other transform parameters can be be coerced into n-dimensions too), though I've only tested for n=2. I think this should be compatible with ITK's origin and direction properties, which seems to be the main use case here, but let me know if I'm missing something.

I'm still planning to make some significant changes to address issue #2163 (including the desired composition order as is also discussed here), but if the above looks good, would you be OK with closing this issue?

**Comment by sofroniewn** [comment]
> To make this example, I split the 3x3 2D rigid transform matrix into a 2x2 rotation matrix and 2x1 translation vector and passed these through to add_image's translate and rotate arguments.

I think this is sufficient to close this issue. See https://github.com/napari/napari/blob/d310847f0c79f062daadecb8a159e8972c074470/napari/layers/base/base.py#L43

I've now merged #2855 which allows an affine matrix and a rotation matrix to be passed and then composed.

Hopefully this meets @thewtex needs for transforms. If not, might be best to open up a new issue and we can keep improving from there.

Thanks for looking into this @andy-sweet and working on #2855!!

**Comment by thewtex** [comment]
@andy-sweet @sofroniewn awesome, thank you!!
